File: backend/src/main/java/com/guildnet/backend/features/post/dto/PostDetailDTO.java
package com.guildnet.backend.features.post.dto;

import com.guildnet.backend.features.communityProfile.dto.CommunityProfileDTO;
import com.guildnet.backend.features.postComment.dto.PostComentDTO;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class PostDetailDTO {
    private Long id;
    private String title;
    private String content;
    private List<String> tags;
    private List<PostComentDTO> postComment;
    private CommunityProfileDTO communityProfile;
    private Long communityId;
}

File: frontend/src/app/models/post/PostDetailDTO.ts
import { PostComentDTO } from "../postComment/PostComentDTO";
import { CommunityProfileDTO } from "../communityProfile/CommunityProfileDTO";

export interface PostDetailDTO {
  id: number;
  title: string;
  content: string;
  tags: string[];
  postComment: PostComentDTO[];
  communityProfile: CommunityProfileDTO;
  communityId: number;
}
